torch/fx/experimental/proxy_tensor.py

```diff
@@ -757,6 +759,108 @@ def disable_autocast_cache():
         torch.set_autocast_cache_enabled(old_value)
 
 
+class ModuleStackTracer(PythonKeyTracer):
+    """Customized version of PythonKeyTracer that retains module stack
+    information in node.meta["nn_module_stack"].
+
+    FX symbolic trace actually does this already, but it relies on `self.root`
+    being the actual module being traced. Since make_fx traces a lambda of our
+    creation, things don't work properly.
+
+    So for this version we hold onto a reference to the original module
+    (scope_root) and monkeypatch it in when capturing the stack trace.
+
+    TODO: we should probably augment the standard FX APIs to better support how
+    make_fx uses it.
+    """
+
+    def __init__(self, scope_root):
```

Comment:
This code should probably also get some dedicated `make_fx` unit tests in `test/test_proxy_tensor.py`